applications/stepswitch/src/stepswitch_resources.erl

```diff
@@ -344,25 +345,42 @@ maybe_resource_to_endpoints(#resrc{id=Id
                                    ,gateways=Gateways
                                    ,global=Global
                                    ,weight=Weight
+                                   ,proxies=Proxies
                                   }
                             ,Number, JObj, Endpoints) ->
     CallerIdNumber = wh_json:get_value(<<"Outbound-Caller-ID-Number">>,JObj),
     case filter_resource_by_rules(Id, Number, Rules, CallerIdNumber, CallerIdRules) of
         {'ok', Number_Match} ->
             lager:debug("building resource ~s endpoints", [Id]),
-            Updates = [{<<"Global-Resource">>, wh_util:to_binary(Global)}
+            CCVUpdates = [{<<"Global-Resource">>, wh_util:to_binary(Global)}
                        ,{<<"Resource-ID">>, Id}
                       ],
-            [wh_json:set_values([{<<"Name">>, Name}
-                                 ,{<<"Weight">>, Weight}
-                                ]
-                                ,update_ccvs(Endpoint, Updates)
-                               )
-             || Endpoint <- gateways_to_endpoints(Number_Match, Gateways, JObj, [])
-            ] ++ Endpoints;
+            Updates = [{<<"Name">>, Name}
+                       ,{<<"Weight">>, Weight}
+                      ],
+            EndpointList = [update_endpoint(Endpoint, Updates, CCVUpdates)
+                 || Endpoint <- gateways_to_endpoints(Number_Match, Gateways, JObj, [])],
+            maybe_add_proxies(EndpointList, Proxies, []) ++ Endpoints;
```

Comment:
If you Endpoints as the accumulator, you can re-work `maybe_add_proxies/3` to do the appends there.